.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,208 @@
+name: Haskell CI
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -euxo pipefail {0}
+
+on:
+  workflow_call:
+    inputs:
+      build-targets:
+        description: Cabal targets to build, e.g. `lib:my-lib`.
+        type: string
+        required: false
+        default: all
+      cabal:
+        description: Cabal version, e.g., "3.12.0.0"
+        type: string
+        required: false
+        default: latest
+      check:
+        description: Whether to run `cabal check`.
+        type: boolean
+        required: false
+        default: true
+      compat:
+        description: Whether to run the Cabal/GHC compatibility check.
+        type: boolean
+        required: false
+        default: true
+      compat-dirs:
+        description: Directories in which to run the Cabal/GHC compatibility check, relative to the repo root.
+        type: string
+        required: false
+        default: "."
+      configure-flags:
+        description: Flags to pass to `cabal configure`, e.g., `--ghc-options='-Werror=default -j'`
+        type: string
+        required: false
+        # For an explanation of `--ghc-options='-j'`, see Note [Parallelism].
+        default: --ghc-options='-j' --enable-tests
+      ghc:
+        description: GHC version, e.g., "9.10.1"
+        type: string
+        required: false
+        default: latest
+      haddock:
+        description: Whether to run `cabal haddock`.
+        type: boolean
+        required: false
+        default: true
+      os:
+        description: |
+          Runner OS, e.g., "ubuntu-latest".
+
+          Set this to the value of `runs-on`, *not* `runner.os`.
+        type: string
+        required: true
+      sdist:
+        description: |
+          Whether to run `cabal sdist`.
+
+          This will also cause later steps to be run in the unpacked sdist.
```

Comment:
Maybe add a little more context to the description here: "This is useful to verify that the cabal file is total with regards to the set of files (data files, doc files, etc.) because the unpacked sdist will only contain files specified in the cabal file."